File: Data Structure and Algorithm/Assignment 03/Program 4/prog4.c
// Write a C program to store polynomial using linked list. Store the term in ascending
// order.

#include<stdio.h>
#include<stdlib.h>

//Make a structure for making a node for every operands of the polynomial
typedef struct Node{
    int coeff;
    int power;
    struct Node * next;
}node;

//Arrangement the node 
node *createNode(){
    node *head = NULL, *nw;
    int pow, coef, ch=1;
    while(ch != 0){
        printf("Enter the coefficient of the polynominal: ");
        scanf("%d",&coef);
        printf("Enter the power of the coefficient: ");
        scanf("%d",&pow);

        nw = (node*)malloc(sizeof(node));
        nw->coeff = coef;
        nw->power = pow;
        nw->next = NULL;

        if(head == NULL || head->power > pow){
            nw->next = head;
            head = nw;
        } 
        else{
            node *p = head;
            while (p->next!= NULL && p->next->power <pow)
            {
                p = p->next;
            }
            nw->next = p->next;
            p->next = nw;
            
        }

        printf("Do you want to insert: ");
        scanf("%d",&ch);
    }
    
    return head;
}

void display(node *head){
    node *p = head;
    while (p != NULL){
        printf("%d x^(%d)+",p->coeff,p->power);
        p = p->next;
    }
    printf("0");
}

int main(){
    node *h ;
    h = createNode();
    display(h);
    return 0;
}

File: Assignment 03/Program 4/prog4.py
# Write a C program to store polynomial using linked list. Store the term in ascending
# order.

class node:
    def __init__(self,coefficient,power):
        self.coefficient = coefficient
        self.power = power
        self.next = None
    
class linkedList:
    def __init__(self):
        self.head = None
    
    def createNode(self,coefficient,power):
        new_node = node(coefficient,power)

        if self.head == None or power < self.head.power:
            new_node.next = self.head
            self.head = new_node
        else:
            p = self.head
            while p.next != None and p.next.power <power:
                p = p.next
            new_node.next = p.next
            p.next = new_node

    def display(self):
        p =  self.head
        while p :
            print(f"{p.coefficient}x^{p.power}", end=" + ")
            p = p.next
        print("None")


if __name__ == "__main__":
    poly = linkedList()

    ch = -1
    while(ch != 0):
        coef = int(input("Enter the coffecient: "))
        power = int(input("Enter the power: "))
        poly.createNode(coef,power)
        ch = int(input("Do you want to innsert: "))
    poly.display()